examples/api/lib/widgets/scroll_notification_observer/scroll_notification_observer.0.dart

```diff
@@ -0,0 +1,136 @@
+
+class ScrollNotificationObserverApp extends StatelessWidget {
+  const ScrollNotificationObserverApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(useMaterial3: true),
+      // The Scaffold widget contains a [ScrollNotificationObserver].
+      // This is used by [AppBar] for its scrolled under behavior.
+      //
+      // We can use [ScrollNotificationObserver.maybeOf] to get the
+      // state of this [ScrollNotificationObserver] from descendants
+      // of the Scaffold widget.
+      //
+      // If you're not using a [Scaffold] widget, you can create a  [ScrollNotificationObserver]
+      // to notify its descendants of scroll notifications by adding it to the subtree.
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('ScrollNotificationObserver Sample'),
+        ),
+        body: const ScrollNotificationObserverExample(),
+      ),
+    );
+  }
+}
+
+class ScrollNotificationObserverExample extends StatefulWidget {
+  const ScrollNotificationObserverExample({super.key});
+
+  @override
+  State<ScrollNotificationObserverExample> createState() => _ScrollNotificationObserverExampleState();
+}
+
+class _ScrollNotificationObserverExampleState extends State<ScrollNotificationObserverExample> {
+  ScrollNotificationObserverState? _scrollNotificationObserver;
+  ScrollController controller = ScrollController();
+  bool _scrolledDown = false;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    // Remove any previous listener.
+    _scrollNotificationObserver?.removeListener(_handleScrollNotification);
+    // Get the ScrollNotificationObserverState from the Scaffold widget.
+    _scrollNotificationObserver = ScrollNotificationObserver.maybeOf(context);
+    // Add a new listener.
+    _scrollNotificationObserver?.addListener(_handleScrollNotification);
+  }
+
+  @override
+  void dispose() {
+    if (_scrollNotificationObserver != null) {
+      _scrollNotificationObserver!.removeListener(_handleScrollNotification);
+      _scrollNotificationObserver = null;
+    }
+    controller.dispose();
+    super.dispose();
+  }
+
+  void _handleScrollNotification(ScrollNotification notification) {
+    // Check if the notification is a scroll update notification and if the
+    // `notification.depth` is 0. This way we only listen to the scroll
+    // notifications from the closest scrollable, instead of those that may be nested.
+    if (notification is ScrollUpdateNotification && defaultScrollNotificationPredicate(notification)) {
+      final bool oldScrolledDown = _scrolledDown;
+      final ScrollMetrics metrics = notification.metrics;
+      // `_scrolledDown` will be true if the user is scrolling down.
+      _scrolledDown = metrics.extentBefore > 0;
+      // Only update the state if the scrolled down value changed.
+      if (_scrolledDown != oldScrolledDown) {
+        setState(() {});
```

Comment:
Empty set state calls are an anti pattern, they can make it difficult to debug issues when it is not readily apparent what has actually changed. :)

can you adjust this to (pseudo-ing):

```dart
if (_scrolledDown != metrics.extentBefore > 0) {
  setState(() {
    _scrolledDown = metrics.extentBefore > 0;
  });
}
```